### PR TITLE
Add option to pass compile_kwargs for each player separately

### PR DESCRIPTION
**Pull request by d4nst**
Adding this option would be useful to implement models where the discriminator and generator losses have different weights. 

For example, the BEGAN loss function:

image

could be implemented by passing a symbolic variable `kt` as a loss weight for the discriminator that could be updated in a callback:

 ```
    kt = K.variable(0.0)
    model.adversarial_compile(adversarial_optimizer=AdversarialOptimizerSimultaneous(),
                              player_optimizers=[Adam(1e-4), Adam(1e-4)],
                              player_compile_kwargs=[{}, {'loss_weights': {'yfake': kt}}],
                              loss=custom_loss)
```

### CODE REVIEW

File: keras_adversarial/adversarial_model.py [hunk 1]
```diff
@@ -55,13 +55,14 @@ def __init__(self, player_params, base_model=None, player_models=None, player_na
             assert (len(player_models) == self.player_count)
             self.layers = player_models
 
-    def adversarial_compile(self, adversarial_optimizer, player_optimizers, loss, compile_kwargs={},
+    def adversarial_compile(self, adversarial_optimizer, player_optimizers, loss, player_compile_kwargs=[],
```
Comment (bstriner) [inline keras_adversarial/adversarial_model.py]: Default should be `None`. Bad habit to have default be a mutable.

File: keras_adversarial/adversarial_model.py [hunk 2]
```diff
@@ -74,7 +75,8 @@ def adversarial_compile(self, adversarial_optimizer, player_optimizers, loss, co
         self.optimizer = None
```
Comment (bstriner) [inline keras_adversarial/adversarial_model.py:76]: Add default
```python
if player_compile_kwargs is None:
   player_compile_kwargs = [{} for _ in self.layers]
```

File: keras_adversarial/adversarial_model.py [hunk 3]
```diff
@@ -74,7 +75,8 @@ def adversarial_compile(self, adversarial_optimizer, player_optimizers, loss, co
...
 
         # Build player models
-        for opt, model in zip(self.optimizers, self.layers):
+        for opt, model, compile_kwargs in itertools.zip_longest(
```
Comment (bstriner) [inline keras_adversarial/adversarial_model.py]: can just use regular `zip` if the default is the right length. `zip_longest` is failing the py2 tests on travis.

**Review by bstriner — CHANGES_REQUESTED** [review]
Great addition! Need some tweaks so it will pass all of the travis tests.

**Comment by bstriner** [comment]
Great idea, and if someone wants the old functionality they can just duplicate the kwargs several times. Needs some tweaks to pass travis (see review). I'm away this week if you want to make the changes yourself. Otherwise, I'll try to put this in next weekend.

**Comment by d4nst** [comment]
Agree with your review. Just made a commit with the changes.

**Review by bstriner — APPROVED** [review]

**Comment by bstriner** [comment]
Tests look good. Thanks a lot!
